Kudu.Core/Infrastructure/PathUtils/PathWindowsUtility.cs

```diff
@@ -121,7 +121,7 @@ internal override string ResolveMSBuild15Dir()
             return probPaths.FirstOrDefault(path => Directory.Exists(path));
         }
 
-        internal override string ResolveMSBuild16Dir()
+        internal override string ResolveMSBuild16Dir(bool IsDotNet31Project=false)
```

Comment:
Trying to understand why parameter default value was used? `ResolveMSBuild16Dir` is only referenced twice. Can we just pass the right value from both method calls?

Reply:
This I did to ensure `no regression issue` and making sure that it points to `MSBuild 16.7` only for `dotnet 3.1` project. As I see from SiteBuilderFactory, this path can be taken for another target framework as well (anything not `net5.0` and `net6.0`) hence this default to `false` will ensure it explicitly takes that path only for `3.1` project